### PR TITLE
[unitaryhack] DOC: Hyperlinks to QuTiP functions and classes fixed #1526

### LINKED ISSUE #1526: DOC: Hyperlinks to QuTiP functions and classes
It would be nice if the docstrings in QuTiP were well formatted, such that, when Sphinx builds the API and guidelines documentation, there are working link to QuTiP functions and classes. An example of working link is `qutip.steadystate.steadystate` in
http://qutip.org/docs/latest/guide/guide-steady.html#steady-state-solvers-in-qutip

An example of working link in the API doc is options: `qutip.solver.Options` in the attributes of the  `HEOMSolver` class
http://qutip.org/docs/latest/apidoc/classes.html#non-markovian-solvers

A way to proceed would be to:

1) reformat all instances in the docstrings of qutip/qutip such that they are in the form of :py:meth:`mymodule.MyClass.mymethod`, see
 https://stackoverflow.com/questions/22700606/how-would-i-cross-reference-a-function-generated-by-autodoc-in-sphinx

2) Build the documentation with Sphinx to check that the change works and generates an active hyperlink in the relative html page of the documentation. 

3) Iterate 1) and 2) for instances in which Qobj or other QuTiP functions or classes are found.

### PR DESCRIPTION
**Pull request by spinolino**
**Description**
Hyperlinks to functions and classes in the documentation were fixed/added.

Following classes and functions were added to the documentation, as there were broken-hyperlinks pointing to them:

Functions:
- Density Matrix Metrics:
  - Added: hellinger_dist, unitarity, dnorm
-  Measurement of quantum states:
   - Added: measure_povm, measurement_statistics_povm

Classes:
- Added: Bloch3d

**Related issues or PRs**
fixes #1526

**Changelog**
 DOC: Hyperlinks to QuTiP functions and classes fixed

### CODE REVIEW

File: doc/guide/guide-visualization.rst [hunk 1]
```diff
@@ -388,7 +388,7 @@ Note that to obtain :math:`\chi` with this method we have to construct a matrix
 Implementation in QuTiP
 -----------------------
 
-In QuTiP, the procedure described above is implemented in the function :func:`qutip.tomography.qpt`, which returns the :math:`\chi` matrix given a density matrix propagator. To illustrate how to use this function, let's consider the :math:`i`-SWAP gate for two qubits. In QuTiP the function :func:`qutip.qip.operations.iswap` generates the unitary transformation for the state kets:
+In QuTiP, the procedure described above is implemented in the function :func:`qutip.tomography.qpt`, which returns the :math:`\chi` matrix given a density matrix propagator. To illustrate how to use this function, let's consider the :math:`i`-SWAP gate for two qubits. In QuTiP the function :func:`qutip.qip.operations.iswap<qutip.qip.operations.gates.iswap>` generates the unitary transformation for the state kets:
```
Comment (hodgestar) [inline doc/guide/guide-visualization.rst:391]: Is the `<>` bit correct? It looks like maybe the intention was just to add a `.gates`?

File: doc/guide/qip/qip-basics.rst [hunk 2]
```diff
@@ -326,9 +326,9 @@ There are two different ways to simulate the action of quantum circuits using Qu
...
 - The first method utilizes unitary application through matrix products on the input states.
   This method simulates circuits exactly in a deterministic manner. This is achieved through
-  :class:`~qutip.qip.CircuitSimulator`. A short guide to exact simulation can be
+  :class:`~qutip.qip.circuit.CircuitSimulator`. A short guide to exact simulation can be
   found at :ref:`qip_simulator`. The teleportation notebook is also useful as an example.
 
 - A different method of circuit simulation employs driving Hamiltonians with the ability to
   simulate circuits in the presence of noise. This can be achieved through the various classes
-  in :class:`~qutip.qip.device`.A short guide to processors for QIP simulation can be found at :ref:`qip_processor`.
+  in :class:`qutip.qip.device`.A short guide to processors for QIP simulation can be found at :ref:`qip_processor`.
```
Comment (hodgestar) [inline doc/guide/qip/qip-basics.rst:334]:
```suggestion
  in :class:`qutip.qip.device`. A short guide to processors for QIP simulation can be found at :ref:`qip_processor`.
```

**Review by hodgestar — CHANGES_REQUESTED** [review]
Left a couple of small suggestions, but looks great.

**Comment by hodgestar** [comment]
CodeClimate reported a bunch of places where line lengths were increased to exceed 79 characters. Could we reduce those down again if possible? Maybe do this at the end so that it's not too hard to review the reformatting.

File: doc/guide/guide-visualization.rst (hunk 1 above)
Comment (spinolino) [reply doc/guide/guide-visualization.rst:391]: The method iswap works both ways (with or without .gates). So I didn't want to change the shorter version, but had to add a working link to this function. So I used the sphinx "text-to-link" syntax:
`:func:'Text <link>'`.

Comment (hodgestar) [reply doc/guide/guide-visualization.rst:391]: Cool. Let's leave it like this then.

**Review by hodgestar — APPROVED** [review]
Thanks @spinolino!

**Pull request merged**